common/data/src/main/java/org/thingsboard/mqtt/broker/common/data/client/credentials/HasSinglePubSubAutorizationRules.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.mqtt.broker.common.data.client.credentials;
+
+public interface HasSinglePubSubAutorizationRules {
```

Comment:
you have a typo here: not HasSinglePubSubAutorizationRules but HasSinglePubSubAuthorizationRules.

Maybe the **SinglePubSubAuthRulesAware** name would be better?